src/object/srv_obj.c

```diff
@@ -2863,6 +2863,22 @@ obj_local_enum(struct obj_io_context *ioc, crt_rpc_t *rpc,
 		}
 	}
 
+	if ((rc == -DER_KEY2BIG) && opc == DAOS_OBJ_RPC_ENUMERATE &&
+	    enum_arg->kds_len < 4) {
+		/* let's query the total size for one update (oid/dkey/akey/rec)
+		 * to make sure the migration/enumeration can go ahead.
+		 */
+		enum_arg->size_query = 1;
+		enum_arg->kds_len = 0;
+		enum_arg->kds[0].kd_key_len = 0;
+		enum_arg->kds_cap = 4;
```

Comment:
not very sure, just confirm how do you know it is safe to set kds_cap as 4? is it possibly the valid kds[] array is with less length? may add an assertion?

Reply:
oh, sure. but it is impossible for kds number < 4 for migration.